### PR TITLE
Add developer documentation.

### PR DESCRIPTION
**Pull request by newswangerd**
#### What is this PR doing:

Add user and developer documentation.

Note to the reviewers, this adds an excalidraw file for the diagrams, which is where 15,000 new lines come from

### CODE REVIEW

**Comment by alikins** [comment]
Nice! :wave:

**Comment by bmbouter** [comment]
Wow these docs are nice! :100:

File: docs/dev/developer_guide/overview.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Overview
+
+## Key Technologies
+
+Galaxy NG relies on the following dependencies:
+
+- [Pulp](https://pulpproject.org/): software repository (more on this bellow)
```
Comment (ShaiahWren) [inline docs/dev/developer_guide/overview.md:7]: below (not 'bellow') https://proofed.com/writing-tips/word-choice-below-vs-bellow/#:~:text=Below%20is%20a%20preposition%20or,level%2C%20layer%2C%20or%20rank.&text=Bellow%20is%20a%20verb%20that,tool%20used%20to%20strengthen%20fire.

Comment (ShaiahWren) [inline docs/dev/developer_guide/overview.md:7]: below (not bellow) see https://proofed.com/writing-tips/word-choice-below-vs-bellow/#:~:text=Below%20is%20a%20preposition%20or,level%2C%20layer%2C%20or%20rank.&text=Bellow%20is%20a%20verb%20that,tool%20used%20to%20strengthen%20fire.

File: docs/dev/developer_guide/overview.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+- [Postgres](https://www.postgresql.org/): database.
+- [Dynaconf](https://www.dynaconf.com/): python settings library. Used to manage configurations.
+
+Of this list, the most important one to understand for the purpose of backend development is Pulp, and the rest of this document will focus on that. For the rest of the technologies in the stack, please refer to their respective documentation.
+
+## An overview of Pulp
+
+Pulp is a pluggable library for managing software packages. It has the ability to store software (such as ansible collections and container images), sync them from remote sources (such as galaxy.ansible.com and quay.io) and organize them into repositories. We make use of the following pulp projects:
+
+- [pulpcore](https://github.com/pulp/pulpcore/): by itself, this doesn't do a whole lot it provides the generic data structures required for storing any software type, which are added using plugins.
```
Comment (ShaiahWren) [inline docs/dev/developer_guide/overview.md:20]:
```suggestion
- [pulpcore](https://github.com/pulp/pulpcore/): by itself, this doesn't do a whole lot. It provides the generic data structures required for storing any software type, which are added using plugins.
```

File: docs/dev/developer_guide/data_models.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+**Example**
+
+<!-- Diagram source can be found in diagrams/model-diagrams.excalidraw and can be edited on https://excalidraw.com/ -->
+![Content/Artifact Relationship](diagrams/content-artifact-diagram.png)
+
+### Repository
+
+A big bucket of content. Think of this like a folder that has a set of collections (or any other pulp content) in it. Repositories store pointers to content, not the actual content itself.
+
+Repositories are versioned and repository versions are immutable. What this means is that whenever content is added or removed from a repository a new repository version is created that references the list of changes that happened on the repository. This is similar to how git works where code in a git repository is a long list of commits that reference changes that were made to the code over time. Versions allow for repositories to be rolled back to a previous state
```
Comment (ShaiahWren) [inline docs/dev/developer_guide/data_models.md:28]:
```suggestion
Repositories are versioned and repository versions are immutable. What this means is that whenever content is added or removed from a repository a new repository version is created that references the list of changes that happened on the repository. This is similar to how git works where code in a git repository is a long list of commits that reference changes that were made to the code over time. Versions allow for repositories to be rolled back to a previous state.
```

File: docs/dev/developer_guide/data_models.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+The repository object in the database points to whatever the latest repository version is, however content can be referenced in any version of a repository, not just the latest version.
```
Comment (ShaiahWren) [inline docs/dev/developer_guide/data_models.md:30]:
```suggestion
The repository object in the database points to whatever the latest repository version is; however, content can be referenced in any version of a repository, not just the latest version.
```

File: docs/dev/developer_guide/data_models.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+- Remote
+    - RoleRemote: Server that supports the v1 galaxy API. Currently unused.
+    - CollectionRemote: Server that supports either the v2 or v3 galaxy API.
+    - GitRemote: Git server that contains a collection as a git repository. Currently unused.
+- Repository
+    - AnsibleRepository: contains all the ansible content types listed above.
+- Distribution
+    - AnsibleDistrbution
+
+Additionally Pulp Ansible defines the following models that don't extend any of the pulp base classes:
```
Comment (ShaiahWren) [inline docs/dev/developer_guide/data_models.md:92]:
```suggestion
Additionally, Pulp Ansible defines the following models that don't extend any of the pulp base classes:
```

File: docs/dev/developer_guide/data_models.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+- Namespace: Contains information about the publisher of a collection. This also defines who can publish collections within a given namespace.
+
+**Example**
+
+<!-- Diagram source can be found in diagrams/model-diagrams.excalidraw and can be edited on https://excalidraw.com/ -->
+![Pulp Ansible Example](diagrams/pulp-ansible.png)
+
+## Pulp Container Content Model
+
+The pulp container data model is a little bit counter-intuitive and bears some additional explanation. For this example lets assume we have a container registry running on my-registry.example.com with the following containers:
```
Comment (ShaiahWren) [inline docs/dev/developer_guide/data_models.md:104]:
```suggestion
The pulp container data model is a little bit counter-intuitive and merits some additional explanation. For this example lets assume we have a container registry running on my-registry.example.com with the following containers:
```

**Review by rochacbruno — APPROVED** [review]

**Review by jerabekjiri — APPROVED** [review]

**Pull request merged**